### PR TITLE
UPSTREAM: <carry>: Bump github.com/argoproj/argo-workflows/v3 from v3.3.10 to v3.5.8

### PR DESCRIPTION
**Pull request by hbelmiro**
**Description of your changes:**
Fixes the `github.com/satori/go.uuid` CVE from https://issues.redhat.com/browse/RHOAIENG-8495.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 1 commits between dadfb38393ce96cb7dc1eac60428d66f0e3f1933...98295f35769bb41530c8889a56752119b205aa53

UPSTREAM commit 98295f3 has invalid summary Bump github.com/argoproj/argo-workflows/v3 from v3.3.10 to v3.5.8.

UPSTREAM commits are validated against the following regular expression:
  ^UPSTREAM: (revert: )?(([\w.-]+/[\w-.-]+)?: )?(\d+:|<carry>:|<drop>:)

UPSTREAM commit summaries should look like:

  UPSTREAM: <PR number|carry|drop>: description

UPSTREAM commits which revert previous UPSTREAM commits should look like:

  UPSTREAM: revert: <normal upstream format>

Examples of valid summaries:

  UPSTREAM: 12345: A kube fix
  UPSTREAM: <carry>: A carried kube change
  UPSTREAM: <drop>: A dropped kube change
  UPSTREAM: revert: 12345: A kube revert


```

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 1 commits between dadfb38393ce96cb7dc1eac60428d66f0e3f1933...cfaec1f5ac30aff94975018e9728a7453cf756fc

UPSTREAM commit cfaec1f has invalid summary Bump github.com/argoproj/argo-workflows/v3 from v3.3.10 to v3.5.8.

UPSTREAM commits are validated against the following regular expression:
  ^UPSTREAM: (revert: )?(([\w.-]+/[\w-.-]+)?: )?(\d+:|<carry>:|<drop>:)

UPSTREAM commit summaries should look like:

  UPSTREAM: <PR number|carry|drop>: description

UPSTREAM commits which revert previous UPSTREAM commits should look like:

  UPSTREAM: revert: <normal upstream format>

Examples of valid summaries:

  UPSTREAM: 12345: A kube fix
  UPSTREAM: <carry>: A carried kube change
  UPSTREAM: <drop>: A dropped kube change
  UPSTREAM: revert: 12345: A kube revert


```

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-58`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-58`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-58`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-58`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-58`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-58`

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 1 commits between dadfb38393ce96cb7dc1eac60428d66f0e3f1933...8e1d7ac69575df507d03f5e38cd8fc93f6230942

UPSTREAM commit 8e1d7ac has invalid summary Bump github.com/argoproj/argo-workflows/v3 from v3.3.10 to v3.5.8.

UPSTREAM commits are validated against the following regular expression:
  ^UPSTREAM: (revert: )?(([\w.-]+/[\w-.-]+)?: )?(\d+:|<carry>:|<drop>:)

UPSTREAM commit summaries should look like:

  UPSTREAM: <PR number|carry|drop>: description

UPSTREAM commits which revert previous UPSTREAM commits should look like:

  UPSTREAM: revert: <normal upstream format>

Examples of valid summaries:

  UPSTREAM: 12345: A kube fix
  UPSTREAM: <carry>: A carried kube change
  UPSTREAM: <drop>: A dropped kube change
  UPSTREAM: revert: 12345: A kube revert


```

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-58`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-58`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-58`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-58`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-58`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-58`

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 1 commits between dadfb38393ce96cb7dc1eac60428d66f0e3f1933...228cda290ecc41f254074709e83421aecb6e3943

UPSTREAM commit 228cda2 has invalid summary Bump github.com/argoproj/argo-workflows/v3 from v3.3.10 to v3.5.8.

UPSTREAM commits are validated against the following regular expression:
  ^UPSTREAM: (revert: )?(([\w.-]+/[\w-.-]+)?: )?(\d+:|<carry>:|<drop>:)

UPSTREAM commit summaries should look like:

  UPSTREAM: <PR number|carry|drop>: description

UPSTREAM commits which revert previous UPSTREAM commits should look like:

  UPSTREAM: revert: <normal upstream format>

Examples of valid summaries:

  UPSTREAM: 12345: A kube fix
  UPSTREAM: <carry>: A carried kube change
  UPSTREAM: <drop>: A dropped kube change
  UPSTREAM: revert: 12345: A kube revert


```

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-58`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-58`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-58`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-58`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-58`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-58`

**Comment by rimolive** [comment]
@hbelmiro Please follow the suggestions from commit checker comments.

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 1 commits between dadfb38393ce96cb7dc1eac60428d66f0e3f1933...38ce527bd570d2297f8497353944c813655571c2
```

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-58`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-58`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-58`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-58`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-58`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-58`

**Comment by hbelmiro** [comment]
> @hbelmiro Please follow the suggestions from commit checker comments.

@rimolive fixed.

File: backend/src/apiserver/template/argo_template.go [hunk 1]
```diff
@@ -230,7 +230,7 @@ func ValidateWorkflow(template []byte) (*util.Workflow, error) {
 	if wf.Kind != argoK8sResource {
 		return nil, util.NewInvalidInputError("Unexpected resource type. Expected: %v. Received: %v", argoK8sResource, wf.Kind)
 	}
```
Comment (DharmitD) [inline backend/src/apiserver/template/argo_template.go:232]: @gmfrasca could you take a look at these argo code changes please?

**Comment by DharmitD** [comment]
go.mod/go.sum and licenses' updates lgtm, I'll let @gmfrasca / @hbelmiro look at argo related code changes, and approve.

File: backend/src/apiserver/template/argo_template.go (hunk 1 above)
Comment (hbelmiro) [reply backend/src/apiserver/template/argo_template.go:232]: This is due to a breaking change in the Argo's API. `ValidateWorkflow` returns only `error` in the new version.

**Review by DharmitD — COMMENTED** [review]
Package and code changes

/lgtm

@gmfrasca @HumairAK could you make sure to check the argo code changes once before approving? Ty!

File: backend/src/apiserver/template/argo_template.go (hunk 1 above)
Comment (gregsheremeta) [reply backend/src/apiserver/template/argo_template.go:232]: lgtm

**Review by gregsheremeta — COMMENTED** [review]
/lgtm

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 0 commits between 4a147666f16dfabfa49f7574fe604a5364fc5035...38ce527bd570d2297f8497353944c813655571c2
```

**Pull request merged**